legion/engine/physics/systems/physicssystem.cpp

```diff
@@ -3,24 +3,40 @@
 
 namespace legion::physics
 {
+    namespace detail
+    {
+        static inline bool greater_penetration(const physics_contact& contact1, const physics_contact& contact2)
+        {
+            auto dot1 = math::dot(contact1.RefWorldContact - contact1.IncWorldContact, -contact1.collisionNormal);
+            auto dot2 = math::dot(contact2.RefWorldContact - contact2.IncWorldContact, -contact2.collisionNormal);
+            return dot1 < dot2;
+        }
+    }
+
     std::unique_ptr<BroadPhaseCollisionAlgorithm> PhysicsSystem::m_broadPhase = nullptr;
 
-    bool PhysicsSystem::IsPaused = false;
+    bool PhysicsSystem::IsPaused = true;
     bool PhysicsSystem::oneTimeRunActive = false;
 
 
     void PhysicsSystem::setup()
     {
-        createProcess<&PhysicsSystem::fixedUpdate>("Physics", m_timeStep);
+        m_broadPhase = std::make_unique<BroadphaseUniformGridNoCaching>(math::vec3(3, 3, 3));
+    }
 
-        manifoldPrecursorQuery = createQuery<position, rotation, scale, physicsComponent>();
+    void PhysicsSystem::update(legion::time::span deltaTime)
+    {
+        m_accumulator += deltaTime;
 
-        //std::make_unique<BroadphaseUniformGrid>(math::vec3(2,2,2),1);
-        //std::make_unique<BroadphaseBruteforce>();
-        //std::make_unique<BroadphaseUniformGridNoCaching>(math::vec3(2, 2, 2));
+        size_type currentInterval = 0;
 
-        m_broadPhase = std::make_unique<BroadphaseUniformGridNoCaching>(math::vec3(2, 2, 2));
+        while (m_accumulator > m_timeStep && currentInterval < m_maxInterval)
+        {
+            m_accumulator -= m_timeStep;
+            fixedUpdate(m_timeStep);
 
+            currentInterval++;
+        }
```

Comment:
wait why exactly did we stop using the fixed interval process?

Reply:
Initially, this was because there seemed to be something wrong with createProcess causing it to call fixedUpdate in the wrong intervals. I looked at this section of the code again and did some tests, apparently, it was something entirely unrelated (my bad!). But yeah good catch, will revert this.

Reply:
If you would like to change the behaviour of fixed interval process, please let me know. physics was the main motivation behind it's existence. (and also the only that uses it rn) So it's fine to say add the maximum updates per frame that you added here.

Reply:
no what i meant is: you're allowed to go into system and scheduling to change how fixed interval process works. admittedly i didn't spend all too much research into how it should or shouldn't behave.

Reply:
alright, but this is not something I need to do right now since it works as intended for now